### ISSUE TITLE
StreamKafkaPTest.when_partitionAddedWhileJobDown_then_consumedFromBeginning

### ISSUE THREAD

**Issue by mmedenjak** [hazelcast/hazelcast#18470]
http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/4412

```
16:35:26 [ERROR]   StreamKafkaPTest.when_partitionAddedWhileJobDown_then_consumedFromBeginning:407->HazelcastTestSupport.assertTrueEventually:1350->HazelcastTestSupport.assertTrueEventually:1248->lambda$when_partitionAddedWhileJobDown_then_consumedFromBeginning$8:408 expected:<1=1> but was:<0=0>
```

**Comment by mmedenjak** [comment]
@viliam-durina can you take a look?

**Comment by mmedenjak** [comment]
http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/4652

**Comment by viliam-durina** [comment]
The case in https://github.com/hazelcast/hazelcast/issues/18470#issuecomment-820931110 fixed in #18554. The other case is likely a different cause.

**Comment by viliam-durina** [comment]
Had a closer look and the cause is likely similar for both :). Both fixed in #18554
